src/main/java/duke/TaskFinder.java

```diff
@@ -0,0 +1,228 @@
+            }
+        }
+        return toDosNotDone;
+    }
+
+    /**
+     * Searches through task list and
+     * returns an array list of all tasks with a deadline.
+     *
+     * @param taskList list of tasks to filter
+     * @return ArrayList<Task> list of done tasks
+     */
+    public ArrayList<Task> findDeadlines(ArrayList<Task> taskList) {
+        ArrayList<Task> deadlines = new ArrayList<Task>();
+        for (Task task : taskList) {
+            if (task.getClass().getSimpleName().equals("Deadline")) {
+                deadlines.add(task);
+            }
+        }
+        return deadlines;
+    }
+
+    /**
+     * Searches through task list and
+     * returns an array list of all tasks with a deadline
+     * that were marked as done.
+     *
+     * @param taskList list of tasks to filter
+     * @return ArrayList<Task> list of done tasks
+     */
+    public ArrayList<Task> findDeadlinesDone(ArrayList<Task> taskList) {
+        ArrayList<Task> deadlinesDone = new ArrayList<Task>();
+        for (Task task : taskList) {
+            if (task.getClass().getSimpleName().equals("Deadline")
+                    && task.isDone) {
+                deadlinesDone.add(task);
+            }
+        }
+        return deadlinesDone;
+    }
+
+    /**
+     * Searches through task list and
+     * returns an array list of all tasks with a deadline
+     * that were not marked as done.
+     *
+     * @param taskList list of tasks to filter
+     * @return ArrayList<Task> list of done tasks
+     */
+    public ArrayList<Task> findDeadlinesNotDone(ArrayList<Task> taskList) {
+        ArrayList<Task> deadlinesNotDone = new ArrayList<Task>();
+        for (Task task : taskList) {
+            if (task.getClass().getSimpleName().equals("Deadline")
+                    && !task.isDone) {
+                deadlinesNotDone.add(task);
+            }
+        }
+        return deadlinesNotDone;
+    }
+
+    /**
+     * Searches through task list and
+     * returns an array list of all events.
+     *
+     * @param taskList list of tasks to filter
+     * @return ArrayList<Task> list of done tasks
+     */
+    public ArrayList<Task> findEvents(ArrayList<Task> taskList) {
+        ArrayList<Task> events = new ArrayList<Task>();
+        for (Task task : taskList) {
+            if (task.getClass().getSimpleName().equals("Event")) {
+                events.add(task);
+            }
+        }
+        return events;
+    }
+
+    /**
+     * Searches through task list and
+     * returns an array list of all events
+     * that were marked as done.
+     *
+     * @param taskList list of tasks to filter
+     * @return ArrayList<Task> list of done tasks
+     */
+    public ArrayList<Task> findEventsDone(ArrayList<Task> taskList) {
+        ArrayList<Task> eventsDone = new ArrayList<Task>();
+        for (Task task : taskList) {
+            if (task.getClass().getSimpleName().equals("Event")
+                    && task.isDone) {
```

Comment:
Perhaps it could be indented 8 spaces more than the parent line?

Reply:
Which line are you referring to specifically?

Reply:
Apologies, on further thought I think the indentation adheres to the coding standards.